gossipd/gossip.c

```diff
@@ -1939,10 +1939,27 @@ static struct io_plan *conn_init(struct io_conn *conn, struct reaching *reach)
 static struct io_plan *conn_proxy_init(struct io_conn *conn,
 				       struct reaching *reach)
 {
-	assert(reach->addr.itype == ADDR_INTERNAL_WIREADDR);
+	char *host = NULL;
+	u16 port;
+
+	switch (reach->addr.itype) {
+	case ADDR_INTERNAL_WIREADDR:
+		host = fmt_wireaddr_without_port(tmpctx,
+						 &reach->addr.u.wireaddr);
+		port = reach->addr.u.wireaddr.port;
+		break;
+	case ADDR_INTERNAL_SOCKNAME:
+	case ADDR_INTERNAL_ALLPROTO:
+	case ADDR_INTERNAL_AUTOTOR:
+		break;
```

Comment:
`abort();` instead?

Reply:
The next line did that and more, by catching any invalid values and calling status_failed...